modules/sor/sor.service.ts

```diff
@@ -0,0 +1,60 @@
+import { GqlSorGetSwapsResponseNew, GqlSorSwapType } from '../../schema';
+import { networkContext } from '../network/network-context.service';
+import { sorV2Service } from './sorV2/sorV2.service';
+import { prisma } from '../../prisma/prisma-client';
+
+export interface GetSwapsInput {
+    tokenIn: string;
+    tokenOut: string;
+    swapType: GqlSorSwapType;
+    swapAmount: string;
+}
+
+export class SorService {
+    public async getSwaps({
+        tokenIn,
+        tokenOut,
+        swapType,
+        swapAmount,
+    }: GetSwapsInput): Promise<GqlSorGetSwapsResponseNew> {
+        const timestamp = Math.floor(Date.now() / 1000);
+
+        // TODO - SORV1 result - via API call to current API or using SORV1/pools directly?
+        const sorV1Result = 'TODO';
+
+        const sorV2Result = await sorV2Service.getSwaps({
+            tokenIn,
+            tokenOut,
+            swapType,
+            swapAmount,
+        });
+
+        let isSorV1 = false;
+        // TODO - Compare V1 vs V2 result and return/log best
+
+        // Update db with best result so we can track performace
+        await prisma.prismaTradeResult.create({
```

Comment:
Does this seem like an ok way to do this?

Reply:
yep looks good. only thing you could do is use composite keys instead of concated string keys, but that's taste in the end.

Reply:
in the end, I'm not sure we want to save all the SOR queries into the DB. There are a shitload of queries sometimes. I think we can only persist a certain sample size.

Reply:
@johngrantuk when you use console.log() it will log it to cloudwatch as well. In addition you could also publish a metric so we could see a nice dashboard how often v1 was better vs v2 and then go into the logs to investigate. For this, you would need to create another metrics client similar to this one https://github.com/beethovenxfi/beethovenx-backend/blob/sor-v2/modules/metrics/cron.metric.ts